src/lib/fw_performance_model/PerformanceModel.hpp

```diff
@@ -0,0 +1,156 @@
+	 */
+	float getMaximumClimbRate(float air_density) const;
+
+	/**
+	 * Get the minimum sink rate (true airspeed) expected for current air density and weight.
+	 * @param air_density in kg/m^3
+	 * @return minimum sink rate in m/s
+	 */
+	float getMinimumSinkRate(float air_density) const;
+
+	/**
+	 * Get the ration of actual weight to base weight
+	 * @return weight ratio
+	 */
+	float getWeightRatio() const;
+
+	/**
+	 * Get the trim throttle for the current airspeed setpoint as well as air density and weight.
+	 * @param throttle_min minimum throttle in range [0,1]
+	 * @param throttle_max maximum throttle in range [0,1]
+	 * @param airspeed_sp calibrated airspeed setpoint in m/s
+	 * @param air_density air density in kg/m^3
+	 * @return trim throttle in range [0,1]
+	 */
+	float getTrimThrottle(float throttle_min, float throttle_max, float airspeed_sp, float air_density) const;
+
+	/**
+	 * Get the trim airspeed compensated for weight.
+	 * @return calibrated trim airspeed in m/s
+	 */
+	float getCalibratedTrimAirspeed() const;
+
+	/**
+	 * Get the minimum airspeed compensated for weight and load factor due to bank angle.
+	 * @param load_factor due to banking
+	 * @return calibrated minimum airspeed in m/s
+	 */
+	float getMinimumCalibratedAirspeed(float load_factor = 1.0f) const;
+
+	/**
+	 * Get the maximum airspeed.
+	 * @return calibrated maximum airspeed in m/s
+	 */
+	float getMaximumCalibratedAirspeed() const;
+
+	/**
+	 * get the stall airspeed compensated for load factor due to bank angle.
+	 * @param load_factor load factor due to banking
+	 * @return calibrated stall airspeed in m/s
+	 */
+	float getCalibratedStallAirspeed(float load_factor) const;
+
+	/**
+	 * Run some checks on parameters and detect unfeasible combinations.
+	 * @return true if all checks pass
+	 */
+	bool runSanityChecks() const;
+
+private:
+	DEFINE_PARAMETERS(
+		(ParamFloat<px4::params::FW_AIRSPD_MAX>) _param_fw_airspd_max,
+		(ParamFloat<px4::params::FW_AIRSPD_MIN>) _param_fw_airspd_min,
+		(ParamFloat<px4::params::FW_AIRSPD_TRIM>) _param_fw_airspd_trim,
+		(ParamFloat<px4::params::FW_AIRSPD_STALL>) _param_fw_airspd_stall,
+		(ParamFloat<px4::params::FW_T_CLMB_MAX>) _param_fw_t_clmb_max,
+		(ParamFloat<px4::params::FW_T_SINK_MIN>) _param_fw_t_sink_min,
+		(ParamFloat<px4::params::WEIGHT_BASE>) _param_weight_base,
+		(ParamFloat<px4::params::WEIGHT_GROSS>) _param_weight_gross,
+		(ParamFloat<px4::params::FW_S_CEILING>) _param_service_ceiling,
+		(ParamFloat<px4::params::FW_THR_TRIM>) _param_fw_thr_trim,
+		(ParamFloat<px4::params::FW_THR_IDLE>) _param_fw_thr_idle,
+		(ParamFloat<px4::params::FW_THR_MAX>) _param_fw_thr_max,
+		(ParamFloat<px4::params::FW_THR_MIN>) _param_fw_thr_min,
+		(ParamFloat<px4::params::FW_THR_ASPD_MIN>) _param_fw_thr_aspd_min,
+		(ParamFloat<px4::params::FW_THR_ASPD_MAX>) _param_fw_thr_aspd_max)
+
+	/**
+	 * Get the trim throttle for the current airspeed setpoint.
+	 * @param airspeed_sp in m/s
+	 * @return trim throttle
```

Comment:
```suggestion
	 * Get the sea level trim throttle for a given calibrated airspeed setpoint.
	 * @param airspeed_sp [m/s] calibrated
	 * @return trim throttle [0, 1] at sea level 
```
could also use "calibrated" for throttle - but not sure if that's a correct use of the word.